crates/datagen/src/lib.rs

```diff
@@ -1761,6 +1842,124 @@ impl<'a> RecordGenerator<'a> {
         Ok(())
     }
 
+    fn generate_decimal(
+        &self,
+        field: &Field,
+        settings: &RngFieldSettings,
+        incr: usize,
+        rng: &mut SmallRng,
+        obj: &mut Value,
+    ) -> AnyResult<()> {
+        let (min, max) = decimal_max_range(field)?;
+        let range = range_as_decimal(&field.name, &settings.range)?;
+        if let Some((a, b)) = range {
+            if a > b {
+                return Err(anyhow!(
+                    "Invalid range, min > max for field {:?}",
+                    field.name
+                ));
+            }
+        }
+        if let Some(nl) = Self::maybe_null(field, settings, rng) {
+            *obj = nl;
+            return Ok(());
+        }
+        let scale = Decimal::from(settings.scale);
+        let incr_dec = Decimal::from(incr);
+
+        match (&settings.strategy, &settings.values, &range) {
+            (DatagenStrategy::Increment, None, None) => {
+                let val = (incr_dec * scale) % max;
```

Comment:
I don't understand what happens here.
what is scale?
I think % makes sense only for integers (but perhaps it has some meaning for decimals too?)
I would expect that it's easiest to generate an integer and divide it by 10^scale

Reply:
that's yet another `scale` from datagen settings, to scale the value